source/extensions/common/wasm/wasm.cc

```diff
@@ -930,9 +955,80 @@ void Context::onHttpCallResponse(uint32_t token, const Pairs& response_headers,
                              trailers_ptr, trailers_size);
 }
 
+uint32_t Context::defineMetric(MetricType type, absl::string_view name) {
```

Comment:
We are iterating over all the possible enum value.
Could you use `switch case without default`? That's a strong indication that all the values are considered. And modern compiler can guarantee that (if a value is skipped compiler will fail to compile)

Reply:
Sadly that is only for well behaved code.  There is an underlying enum type and nothing prevents a poorly behaved program from mashing that value at which point the non of the cases are executed:

#include <iostream>

enum class V : int {
  No = 0,
  Yes = 1,
};

V make_v() {
  return static_cast<V>(2);
}

int main() {
  V v = make_v();
  switch (v) {
    case V::No:
      std::cout << "No";
      break;
    case V::Yes:
      std::cout << "Yes";
      break;
  }
  std::cout << "Done";
}

./a.out
Done

Reply:
That was "#include <iostream>"

Reply:
Thank you for this code. I didn't think of this bad value case. However, isn't this a perfect usage in this case?
- return id in known value (and early return)
- replace `cout << "Done"` by `return 0`

Reply:
Changed it to a 'switch'.  Code is much larger because of the additional scopes required to handle the local iterator, but it does mean that there will be an error if a new type is added to the enum.